tests/pipeline/nodes/model/fairmotv1/test_fairmot.py

```diff
@@ -0,0 +1,314 @@
+        {"key": "min_box_area", "value": -0.5},
+        {"key": "track_buffer", "value": -0.5},
+    ],
+)
+def fairmot_negative_config_value(request, fairmot_config):
+    """Various invalid config values."""
+    fairmot_config[request.param["key"]] = request.param["value"]
+    return fairmot_config
+
+
+# def replace_download_weights(*_):
+#     pass
+
+
+def replace_fuse_motion(*args):
+    """Manipulate the computed embedding distance so they are too large and
+    cause none of the detections to be associated. This forces the Tracker to
+    associate with IoU costs.
+    """
+    return np.ones_like(fuse_motion(*args))
+
+
+def replace_iou_distance(*args):
+    """Manipulate the computed IoU-based costs so they are too large and
+    cause none of the detections to be associated. This forces the Tracker to
+    mark tracks for removal.
+    """
+    return np.ones_like(iou_distance(*args))
+
+
+class TestFairMOT:
+    def test_should_give_empty_output_for_no_human_images(
+        self, test_no_human_images, fairmot_config
+    ):
+        """Input images either contain nothing or non-humans."""
+        blank_image = cv2.imread(test_no_human_images)
+        fairmot = Node(fairmot_config)
+        output = fairmot.run({"img": blank_image})
+        expected_output = {
+            "bboxes": [],
+            "bbox_labels": [],
+            "bbox_scores": [],
+            "obj_tags": [],
+        }
+        assert output.keys() == expected_output.keys()
+        npt.assert_equal(output["bboxes"], expected_output["bboxes"])
+        npt.assert_equal(output["bbox_labels"], expected_output["bbox_labels"])
+        npt.assert_equal(output["bbox_scores"], expected_output["bbox_scores"])
+        npt.assert_equal(output["obj_tags"], expected_output["obj_tags"])
+
+    def test_tracking_ids_should_be_consistent_across_frames(
+        self, test_human_video_sequences, fairmot_config
+    ):
+        _, detections = test_human_video_sequences
+        fairmot = Node(fairmot_config)
+        prev_tags = []
+        for i, inputs in enumerate({"img": x["img"]} for x in detections):
+            output = fairmot.run(inputs)
+            if i > 1:
+                for track_id, track in enumerate(fairmot.model.tracker.tracked_stracks):
+                    assert repr(track) == f"OT_{track_id + 1}_(1-{i + 1})"
+                assert output["obj_tags"] == prev_tags
+            prev_tags = output["obj_tags"]
+
+    @pytest.mark.skipif(not torch.cuda.is_available(), reason="requires GPU")
+    def test_tracking_ids_should_be_consistent_across_frames_gpu(
+        self, test_human_video_sequences, fairmot_config_gpu
+    ):
+        _, detections = test_human_video_sequences
+        fairmot = Node(fairmot_config_gpu)
+        prev_tags = []
+        for i, inputs in enumerate({"img": x["img"]} for x in detections):
+            output = fairmot.run(inputs)
+            if i > 1:
+                assert output["obj_tags"] == prev_tags
```

Comment:
How come lines 122-123 are not needed here? Does testing with GPU change anything?

Reply:
The `STrack`'s `track_id` depends on the class variable `_count` in `BaseTrack`. When running CPU and GPU test back to back, the `_count` value is retained and cause the gpu test to fail.

But running the GPU test by itself passes, to avoid having to change the code to suit the test, the `repr` is not tested again in the GPU test.